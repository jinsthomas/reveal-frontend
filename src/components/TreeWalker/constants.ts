export const ACTIVE = 'Active';
export const FIND_BY_ID = 'location/findByJurisdictionIds';
export const FIND_BY_PROPERTIES = 'location/findByProperties';
export const LOCATION = 'location';
export const COULDNT_LOAD_PARENTS = 'Could not load parents';
