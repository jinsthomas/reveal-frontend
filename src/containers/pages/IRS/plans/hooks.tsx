import { SupersetFormData } from '@onaio/superset-connector/dist/types';
import React, { useEffect, useState } from 'react';
import { ActionCreator } from 'redux';
import { SUPERSET_IRS_REPORTING_PLANS_SLICE } from '../../../../configs/env';
import { displayError } from '../../../../helpers/errors';
import supersetFetch from '../../../../services/superset';

interface AsyncSupersetPlans<TData, TAction> {
  superset: typeof supersetFetch;
  data: TData[];
  fetchPlans: ActionCreator<TAction>;
  supersetSlice: string;
  supersetOptions: SupersetFormData | null;
}

export const useAsyncSupersetPlans = <TData, TAction>({
  superset = supersetFetch,
  data = [],
  fetchPlans,
  supersetSlice = SUPERSET_IRS_REPORTING_PLANS_SLICE,
  supersetOptions = null,
}: AsyncSupersetPlans<TData, TAction>) => {
  const [loading, setLoading] = useState<boolean>(false);
  const mounted = React.useRef<boolean>(false);

  /** async function to load the data */
  async function loadData() {
    try {
      setLoading(data.length === 0); // only set loading when there are no plans
      const asyncOperation = supersetOptions
        ? superset(supersetSlice, supersetOptions)
        : superset(supersetSlice);

      await asyncOperation.then((result: TData[]) => fetchPlans(result));
    } catch (e) {
      throw e;
    } finally {
      if (mounted) {
        setLoading(false);
      }
    }
  }

  useEffect(() => {
    mounted.current = true;
    loadData().catch(error => displayError(error));
    return () => {
      mounted.current = false;
    };
  }, []);

  return loading;
};
