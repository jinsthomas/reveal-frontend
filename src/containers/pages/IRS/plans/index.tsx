import reducerRegistry from '@onaio/redux-reducer-registry';
import React from 'react';
import { connect } from 'react-redux';
import { RouteComponentProps } from 'react-router';
import { Store } from 'redux';
import { SUPERSET_IRS_REPORTING_PLANS_SLICE } from '../../../../configs/env';
import { IRS_PLANS } from '../../../../configs/lang';
import { QUERY_PARAM_TITLE, REPORT_IRS_PLAN_URL } from '../../../../constants';
import { getQueryParams, RouteParams } from '../../../../helpers/utils';
import supersetFetch from '../../../../services/superset';
import IRSPlansReducer, {
  fetchIRSPlans,
  GenericPlan,
  makeIRSPlansArraySelector,
  reducerName as IRSPlansReducerName,
} from '../../../../store/ducks/generic/plans';
import { GenericPlanListProps, GenericPlansList } from '../../GenericPlansList';

/** register the IRS plan definitions reducer */
reducerRegistry.register(IRSPlansReducerName, IRSPlansReducer);

/** interface for IRS PlanList props */
interface PlanListProps extends GenericPlanListProps {
  ownProps: any;
}

/** Simple component that loads a privew list of IRS plans */
const IRSPlansList = (props: PlanListProps & RouteComponentProps) => {
  return (
    <div>
      <GenericPlansList {...props} />
    </div>
  );
};

/** Declare default props for IRSPlansList */
const defaultProps: PlanListProps = {
  fetchPlans: fetchIRSPlans,
  ownProps: {},
  pageTitle: IRS_PLANS,
  pageUrl: REPORT_IRS_PLAN_URL,
  plans: [],
  service: supersetFetch,
  supersetReportingSlice: SUPERSET_IRS_REPORTING_PLANS_SLICE,
};

IRSPlansList.defaultProps = defaultProps;

export { IRSPlansList };

/** Connect the component to the store */

/** interface to describe props from mapStateToProps */
interface DispatchedStateProps extends RouteComponentProps<RouteParams> {
  plans: GenericPlan[];
}

/** map state to props */
const mapStateToProps = (
  state: Partial<Store>,
  ownProps: RouteComponentProps<RouteParams>
): DispatchedStateProps => {
  const searchedTitle = getQueryParams(ownProps.location)[QUERY_PARAM_TITLE] as string;
  const plans = makeIRSPlansArraySelector()(state, { plan_title: searchedTitle });

  return {
    ...ownProps,
    plans,
  };
};

/** map dispatch to props */
const mapDispatchToProps = { fetchPlans: fetchIRSPlans };

/** Connected ActiveFI component */
const ConnectedIRSPlansList = connect(mapStateToProps, mapDispatchToProps)(IRSPlansList);

export default ConnectedIRSPlansList;
