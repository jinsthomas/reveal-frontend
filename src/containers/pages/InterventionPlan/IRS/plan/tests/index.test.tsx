import { mount, shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import { createBrowserHistory } from 'history';
import React from 'react';
import { Provider } from 'react-redux';
import { Router } from 'react-router';
import { INTERVENTION_IRS_URL } from '../../../../../../constants';
import { OpenSRPService } from '../../../../../../services/opensrp';
import store from '../../../../../../store';
import * as planDucks from '../../../../../../store/ducks/plans';
import * as fixtures from '../../../../../../store/ducks/tests/fixtures';
import {
  irsPlanDefinition1,
  irsPlanRecord1,
  irsPlanRecordNoJurisdictionIds,
  irsPlanRecordResponse1,
  jurisdictionsById,
  jurisidictionResults,
  locationResults,
  parentlessParentJurisdictionResults,
} from '../../tests/fixtures';
import ConnectedIrsPlan, { IrsPlan } from './..';

jest.mock('../../../../../../configs/env');

const history = createBrowserHistory();

/* tslint:disable-next-line no-var-requires */
const fetch = require('jest-fetch-mock');

describe('containers/pages/IRS/plan', () => {
  beforeEach(() => {
    jest.resetAllMocks();
  });

  afterEach(() => {
    jest.clearAllMocks();
  });

  it('renders without crashing', () => {
    const mock: any = jest.fn();
    const props = {
      history,
      location: mock,
      match: {
        isExact: true,
        params: { id: '1234' },
        path: `${INTERVENTION_IRS_URL}/plan/:id`,
        url: `${INTERVENTION_IRS_URL}/plan/1234`,
      },
      planId: '1234',
    };
    shallow(
      <Router history={history}>
        <IrsPlan {...props} />
      </Router>
    );
  });

  it('renders without crashing when loading plans with invalid jurisdiction ids', () => {
    store.dispatch(
      planDucks.fetchPlanRecords([irsPlanRecordResponse1 as planDucks.PlanRecordResponse])
    );

    fetch.mockResponseOnce(JSON.stringify({}));
    const mock: any = jest.fn();
    const { identifier: id } = irsPlanRecordResponse1;
    const supersetServiceMock: any = jest.fn();
    supersetServiceMock.mockImplementation(async () => [
      jurisdictionsById['0'],
      jurisdictionsById['1A'],
      jurisdictionsById['1B'],
      jurisdictionsById['1Aa'],
      jurisdictionsById['1Ab'],
      jurisdictionsById['1Ba'],
      jurisdictionsById['1Bb'],
    ]);
    const props = {
      history,
      location: mock,
      match: {
        isExact: true,
        params: { id },
        path: `${INTERVENTION_IRS_URL}/plan/:id`,
        url: `${INTERVENTION_IRS_URL}/plan/${id}`,
      },
    };

    const wrapper = mount(
      <Provider store={store}>
        <Router history={history}>
          <ConnectedIrsPlan {...props} />
        </Router>
      </Provider>
    );
    wrapper.unmount();
  });

  it('renders correctly if prop planById is not null', async () => {
    const mock: any = jest.fn();
    const supersetServiceMock = jest.fn(async () => jurisidictionResults);
    const { id } = fixtures.plan1;
    const props = {
      history,
      location: mock,
      match: {
        isExact: true,
        params: { id },
        path: `${INTERVENTION_IRS_URL}/plan/:id`,
        url: `${INTERVENTION_IRS_URL}/plan/${id}`,
      },
      planById: irsPlanRecord1,
      supersetService: supersetServiceMock,
    };
    const mockList = jest.fn();
    OpenSRPService.prototype.list = mockList;
    mockList.mockReturnValueOnce(Promise.resolve(fixtures.organizations));
    const extractPlanRecordResponseFromPlanPayloadMock = jest.spyOn(
      planDucks,
      'extractPlanRecordResponseFromPlanPayload'
    );

    const wrapper = mount(
      <Provider store={store}>
        <Router history={history}>
          <ConnectedIrsPlan {...props} />
        </Router>
      </Provider>
    );
    await new Promise<unknown>(resolve => setImmediate(resolve));
    expect(toJson(wrapper.find('Connect(IrsPlan)'))).toMatchSnapshot();
    expect(extractPlanRecordResponseFromPlanPayloadMock).not.toBeCalled();
    wrapper.unmount();
  });

  it('renders correctly if a jurisidiction parent is parentless', async () => {
    const mock: any = jest.fn();
    const supersetServiceMock = jest.fn(async () => parentlessParentJurisdictionResults);
    const { id } = fixtures.plan1;
    const props = {
      history,
      location: mock,
      match: {
        isExact: true,
        params: { id },
        path: `${INTERVENTION_IRS_URL}/plan/:id`,
        url: `${INTERVENTION_IRS_URL}/plan/${id}`,
      },
      planById: irsPlanRecord1,
      supersetService: supersetServiceMock,
    };
    const mockRead = jest.fn();
    const mockList = jest.fn();
    OpenSRPService.prototype.read = mockRead;
    OpenSRPService.prototype.list = mockList;
    mockRead.mockReturnValueOnce(Promise.resolve(locationResults));
    mockList.mockReturnValueOnce(Promise.resolve(fixtures.organizations));

    const wrapper = mount(
      <Provider store={store}>
        <Router history={history}>
          <ConnectedIrsPlan {...props} />
        </Router>
      </Provider>
    );
    await new Promise<unknown>(resolve => setImmediate(resolve));
    expect(toJson(wrapper.find('Connect(IrsPlan)'))).toMatchSnapshot();
    wrapper.unmount();
  });

  it('renders correctly if planById.plan_jurisdictions_ids is undefined', async () => {
    const mock: any = jest.fn();
    const supersetServiceMock = jest.fn(async () => jurisidictionResults);
    const { id } = fixtures.plan1;
    const props = {
      history,
      location: mock,
      match: {
        isExact: true,
        params: { id },
        path: `${INTERVENTION_IRS_URL}/plan/:id`,
        url: `${INTERVENTION_IRS_URL}/plan/${id}`,
      },
      planById: irsPlanRecordNoJurisdictionIds,
      supersetService: supersetServiceMock,
    };
    const mockList = jest.fn();
    OpenSRPService.prototype.list = mockList;
    mockList.mockReturnValueOnce(Promise.resolve(fixtures.organizations));

    const wrapper = mount(
      <Provider store={store}>
        <Router history={history}>
          <ConnectedIrsPlan {...props} />
        </Router>
      </Provider>
    );
    await new Promise<unknown>(resolve => setImmediate(resolve));
    expect(toJson(wrapper.find('Connect(IrsPlan)'))).toMatchSnapshot();
    wrapper.unmount();
  });

  it('renders IRS Plan page correctly', async () => {
    const mock: any = jest.fn();
    const supersetServiceMock: any = jest.fn(async () => jurisidictionResults);
    const { id } = fixtures.plan1;
    const props = {
      history,
      location: mock,
      match: {
        isExact: true,
        params: { id },
        path: `${INTERVENTION_IRS_URL}/plan/:id`,
        url: `${INTERVENTION_IRS_URL}/plan/${id}`,
      },
      supersetService: supersetServiceMock,
    };

    const mockRead = jest.fn();
    const mockList = jest.fn();
    OpenSRPService.prototype.read = mockRead;
    OpenSRPService.prototype.list = mockList;
    mockRead.mockReturnValueOnce(Promise.resolve(irsPlanDefinition1));
    mockList.mockReturnValueOnce(Promise.resolve(fixtures.organizations));
    const extractPlanRecordResponseFromPlanPayloadMock = jest.spyOn(
      planDucks,
      'extractPlanRecordResponseFromPlanPayload'
    );

    const wrapper = mount(
      <Provider store={store}>
        <Router history={history}>
          <ConnectedIrsPlan {...props} />
        </Router>
      </Provider>
    );
    await new Promise<unknown>(resolve => setImmediate(resolve));
    expect(toJson(wrapper.find('Connect(IrsPlan)'))).toMatchSnapshot();
    expect(extractPlanRecordResponseFromPlanPayloadMock.mock.calls.length).toBe(1);
    expect(mockList.mock.calls.length).toBe(1);
    expect(supersetServiceMock.mock.calls.length).toBe(1);
    wrapper.unmount();
  });
});
