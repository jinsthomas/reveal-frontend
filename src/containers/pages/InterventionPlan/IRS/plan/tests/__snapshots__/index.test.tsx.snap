// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`containers/pages/IRS/plan renders IRS Plan page correctly 1`] = `
<Connect(IrsPlan)
  history={
    Object {
      "action": "POP",
      "block": [Function],
      "createHref": [Function],
      "go": [Function],
      "goBack": [Function],
      "goForward": [Function],
      "length": 1,
      "listen": [Function],
      "location": Object {
        "hash": "",
        "pathname": "/",
        "search": "",
        "state": undefined,
      },
      "push": [Function],
      "replace": [Function],
    }
  }
  location={[MockFunction]}
  match={
    Object {
      "isExact": true,
      "params": Object {
        "id": "ed2b4b7c-3388-53d9-b9f6-6a19d1ffde1f",
      },
      "path": "/intervention/irs/plan/:id",
      "url": "/intervention/irs/plan/ed2b4b7c-3388-53d9-b9f6-6a19d1ffde1f",
    }
  }
  supersetService={
    [MockFunction] {
      "calls": Array [
        Array [
          undefined,
          Object {
            "row_limit": 3000,
          },
        ],
      ],
      "results": Array [
        Object {
          "isThrow": false,
          "value": Promise {},
        },
      ],
    }
  }
>
  <IrsPlan
    allJurisdictionIds={
      Array [
        "3952",
      ]
    }
    assignmentsArray={Array []}
    fetchAllJurisdictionIdsActionCreator={[Function]}
    fetchAssignmentsActionCreator={[Function]}
    fetchJurisdictionsActionCreator={[Function]}
    fetchOrganizationsActionCreator={[Function]}
    fetchPlansActionCreator={[Function]}
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "createHref": [Function],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
    isDraftPlan={null}
    isFinalizedPlan={null}
    isNewPlan={false}
    jurisdictionsById={
      Object {
        "3952": Object {
          "geographic_level": 2,
          "jurisdiction_id": "3952",
          "name": "1B - 3952",
          "parent_id": "1B",
        },
      }
    }
    loadedJurisdictionIds={
      Array [
        "3952",
      ]
    }
    location={[MockFunction]}
    match={
      Object {
        "isExact": true,
        "params": Object {
          "id": "ed2b4b7c-3388-53d9-b9f6-6a19d1ffde1f",
        },
        "path": "/intervention/irs/plan/:id",
        "url": "/intervention/irs/plan/ed2b4b7c-3388-53d9-b9f6-6a19d1ffde1f",
      }
    }
    organizationsById={
      Object {
        "4c506c98-d3a9-11e9-bb65-2a2ae2dbcce4": Object {
          "active": true,
          "id": 3,
          "identifier": "4c506c98-d3a9-11e9-bb65-2a2ae2dbcce4",
          "name": "Demo Team",
        },
        "fcc19470-d599-11e9-bb65-2a2ae2dbcce4": Object {
          "active": true,
          "id": 1,
          "identifier": "fcc19470-d599-11e9-bb65-2a2ae2dbcce4",
          "name": "The Luang",
          "type": Object {
            "coding": Array [
              Object {
                "code": "team",
                "display": "Team",
                "system": "http://terminology.hl7.org/CodeSystem/organization-type",
              },
            ],
          },
        },
      }
    }
    planById={null}
    planId="ed2b4b7c-3388-53d9-b9f6-6a19d1ffde1f"
    supersetService={
      [MockFunction] {
        "calls": Array [
          Array [
            undefined,
            Object {
              "row_limit": 3000,
            },
          ],
        ],
        "results": Array [
          Object {
            "isThrow": false,
            "value": Promise {},
          },
        ],
      }
    }
  >
    <Ripple
      borderColor="#ff5d00"
      borderStyle="solid"
      borderWidth="4px"
      height="64px"
      minHeight="80vh"
      width="64px"
    >
      <div
        className="lds-ripple-parent"
        style={
          Object {
            "minHeight": "80vh",
          }
        }
      >
        <div
          className="lds-ripple"
          style={
            Object {
              "height": "64px",
              "width": "64px",
            }
          }
        >
          <div
            style={
              Object {
                "borderColor": "#ff5d00",
                "borderStyle": "solid",
                "borderWidth": "4px",
              }
            }
          />
          <div
            style={
              Object {
                "borderColor": "#ff5d00",
                "borderStyle": "solid",
                "borderWidth": "4px",
              }
            }
          />
        </div>
      </div>
    </Ripple>
  </IrsPlan>
</Connect(IrsPlan)>
`;

exports[`containers/pages/IRS/plan renders correctly if a jurisidiction parent is parentless 1`] = `
<Connect(IrsPlan)
  history={
    Object {
      "action": "POP",
      "block": [Function],
      "createHref": [Function],
      "go": [Function],
      "goBack": [Function],
      "goForward": [Function],
      "length": 1,
      "listen": [Function],
      "location": Object {
        "hash": "",
        "pathname": "/",
        "search": "",
        "state": undefined,
      },
      "push": [Function],
      "replace": [Function],
    }
  }
  location={[MockFunction]}
  match={
    Object {
      "isExact": true,
      "params": Object {
        "id": "ed2b4b7c-3388-53d9-b9f6-6a19d1ffde1f",
      },
      "path": "/intervention/irs/plan/:id",
      "url": "/intervention/irs/plan/ed2b4b7c-3388-53d9-b9f6-6a19d1ffde1f",
    }
  }
  planById={
    Object {
      "id": "0230f9e8-1f30-5e91-8693-4c993661785e",
      "plan_date": "2019-08-09",
      "plan_effective_period_end": "2019-08-29",
      "plan_effective_period_start": "2019-08-09",
      "plan_fi_reason": "",
      "plan_fi_status": "",
      "plan_id": "0230f9e8-1f30-5e91-8693-4c993661785e",
      "plan_intervention_type": "IRS",
      "plan_jurisdictions_ids": Array [
        "3952",
      ],
      "plan_status": "draft",
      "plan_title": "IRS 2019-08-09",
      "plan_version": "1",
    }
  }
  supersetService={
    [MockFunction] {
      "calls": Array [
        Array [
          undefined,
          Object {
            "row_limit": 3000,
          },
        ],
      ],
      "results": Array [
        Object {
          "isThrow": false,
          "value": Promise {},
        },
      ],
    }
  }
>
  <IrsPlan
    allJurisdictionIds={
      Array [
        "3952",
      ]
    }
    assignmentsArray={Array []}
    fetchAllJurisdictionIdsActionCreator={[Function]}
    fetchAssignmentsActionCreator={[Function]}
    fetchJurisdictionsActionCreator={[Function]}
    fetchOrganizationsActionCreator={[Function]}
    fetchPlansActionCreator={[Function]}
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "createHref": [Function],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
    isDraftPlan={null}
    isFinalizedPlan={null}
    isNewPlan={false}
    jurisdictionsById={
      Object {
        "3952": Object {
          "geographic_level": 2,
          "jurisdiction_id": "3952",
          "name": "1B - 3952",
          "parent_id": "1B",
        },
      }
    }
    loadedJurisdictionIds={
      Array [
        "3952",
      ]
    }
    location={[MockFunction]}
    match={
      Object {
        "isExact": true,
        "params": Object {
          "id": "ed2b4b7c-3388-53d9-b9f6-6a19d1ffde1f",
        },
        "path": "/intervention/irs/plan/:id",
        "url": "/intervention/irs/plan/ed2b4b7c-3388-53d9-b9f6-6a19d1ffde1f",
      }
    }
    organizationsById={
      Object {
        "4c506c98-d3a9-11e9-bb65-2a2ae2dbcce4": Object {
          "active": true,
          "id": 3,
          "identifier": "4c506c98-d3a9-11e9-bb65-2a2ae2dbcce4",
          "name": "Demo Team",
        },
        "fcc19470-d599-11e9-bb65-2a2ae2dbcce4": Object {
          "active": true,
          "id": 1,
          "identifier": "fcc19470-d599-11e9-bb65-2a2ae2dbcce4",
          "name": "The Luang",
          "type": Object {
            "coding": Array [
              Object {
                "code": "team",
                "display": "Team",
                "system": "http://terminology.hl7.org/CodeSystem/organization-type",
              },
            ],
          },
        },
      }
    }
    planById={
      Object {
        "id": "0230f9e8-1f30-5e91-8693-4c993661785e",
        "plan_date": "2019-08-09",
        "plan_effective_period_end": "2019-08-29",
        "plan_effective_period_start": "2019-08-09",
        "plan_fi_reason": "",
        "plan_fi_status": "",
        "plan_id": "0230f9e8-1f30-5e91-8693-4c993661785e",
        "plan_intervention_type": "IRS",
        "plan_jurisdictions_ids": Array [
          "3952",
        ],
        "plan_status": "draft",
        "plan_title": "IRS 2019-08-09",
        "plan_version": "1",
      }
    }
    planId="ed2b4b7c-3388-53d9-b9f6-6a19d1ffde1f"
    supersetService={
      [MockFunction] {
        "calls": Array [
          Array [
            undefined,
            Object {
              "row_limit": 3000,
            },
          ],
        ],
        "results": Array [
          Object {
            "isThrow": false,
            "value": Promise {},
          },
        ],
      }
    }
  >
    <Ripple
      borderColor="#ff5d00"
      borderStyle="solid"
      borderWidth="4px"
      height="64px"
      minHeight="80vh"
      width="64px"
    >
      <div
        className="lds-ripple-parent"
        style={
          Object {
            "minHeight": "80vh",
          }
        }
      >
        <div
          className="lds-ripple"
          style={
            Object {
              "height": "64px",
              "width": "64px",
            }
          }
        >
          <div
            style={
              Object {
                "borderColor": "#ff5d00",
                "borderStyle": "solid",
                "borderWidth": "4px",
              }
            }
          />
          <div
            style={
              Object {
                "borderColor": "#ff5d00",
                "borderStyle": "solid",
                "borderWidth": "4px",
              }
            }
          />
        </div>
      </div>
    </Ripple>
  </IrsPlan>
</Connect(IrsPlan)>
`;

exports[`containers/pages/IRS/plan renders correctly if planById.plan_jurisdictions_ids is undefined 1`] = `
<Connect(IrsPlan)
  history={
    Object {
      "action": "POP",
      "block": [Function],
      "createHref": [Function],
      "go": [Function],
      "goBack": [Function],
      "goForward": [Function],
      "length": 1,
      "listen": [Function],
      "location": Object {
        "hash": "",
        "pathname": "/",
        "search": "",
        "state": undefined,
      },
      "push": [Function],
      "replace": [Function],
    }
  }
  location={[MockFunction]}
  match={
    Object {
      "isExact": true,
      "params": Object {
        "id": "ed2b4b7c-3388-53d9-b9f6-6a19d1ffde1f",
      },
      "path": "/intervention/irs/plan/:id",
      "url": "/intervention/irs/plan/ed2b4b7c-3388-53d9-b9f6-6a19d1ffde1f",
    }
  }
  planById={
    Object {
      "id": "0230f9e8-1f30-5e91-8693-4c993661785e",
      "plan_date": "2019-08-09",
      "plan_effective_period_end": "2019-08-29",
      "plan_effective_period_start": "2019-08-09",
      "plan_fi_reason": "",
      "plan_fi_status": "",
      "plan_id": "0230f9e8-1f30-5e91-8693-4c993661785e",
      "plan_intervention_type": "IRS",
      "plan_status": "draft",
      "plan_title": "IRS 2019-08-09",
      "plan_version": "1",
    }
  }
  supersetService={
    [MockFunction] {
      "calls": Array [
        Array [
          undefined,
          Object {
            "row_limit": 3000,
          },
        ],
      ],
      "results": Array [
        Object {
          "isThrow": false,
          "value": Promise {},
        },
      ],
    }
  }
>
  <IrsPlan
    allJurisdictionIds={
      Array [
        "3952",
      ]
    }
    assignmentsArray={Array []}
    fetchAllJurisdictionIdsActionCreator={[Function]}
    fetchAssignmentsActionCreator={[Function]}
    fetchJurisdictionsActionCreator={[Function]}
    fetchOrganizationsActionCreator={[Function]}
    fetchPlansActionCreator={[Function]}
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "createHref": [Function],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
    isDraftPlan={null}
    isFinalizedPlan={null}
    isNewPlan={false}
    jurisdictionsById={
      Object {
        "3952": Object {
          "geographic_level": 2,
          "jurisdiction_id": "3952",
          "name": "1B - 3952",
          "parent_id": "1B",
        },
      }
    }
    loadedJurisdictionIds={
      Array [
        "3952",
      ]
    }
    location={[MockFunction]}
    match={
      Object {
        "isExact": true,
        "params": Object {
          "id": "ed2b4b7c-3388-53d9-b9f6-6a19d1ffde1f",
        },
        "path": "/intervention/irs/plan/:id",
        "url": "/intervention/irs/plan/ed2b4b7c-3388-53d9-b9f6-6a19d1ffde1f",
      }
    }
    organizationsById={
      Object {
        "4c506c98-d3a9-11e9-bb65-2a2ae2dbcce4": Object {
          "active": true,
          "id": 3,
          "identifier": "4c506c98-d3a9-11e9-bb65-2a2ae2dbcce4",
          "name": "Demo Team",
        },
        "fcc19470-d599-11e9-bb65-2a2ae2dbcce4": Object {
          "active": true,
          "id": 1,
          "identifier": "fcc19470-d599-11e9-bb65-2a2ae2dbcce4",
          "name": "The Luang",
          "type": Object {
            "coding": Array [
              Object {
                "code": "team",
                "display": "Team",
                "system": "http://terminology.hl7.org/CodeSystem/organization-type",
              },
            ],
          },
        },
      }
    }
    planById={
      Object {
        "id": "0230f9e8-1f30-5e91-8693-4c993661785e",
        "plan_date": "2019-08-09",
        "plan_effective_period_end": "2019-08-29",
        "plan_effective_period_start": "2019-08-09",
        "plan_fi_reason": "",
        "plan_fi_status": "",
        "plan_id": "0230f9e8-1f30-5e91-8693-4c993661785e",
        "plan_intervention_type": "IRS",
        "plan_status": "draft",
        "plan_title": "IRS 2019-08-09",
        "plan_version": "1",
      }
    }
    planId="ed2b4b7c-3388-53d9-b9f6-6a19d1ffde1f"
    supersetService={
      [MockFunction] {
        "calls": Array [
          Array [
            undefined,
            Object {
              "row_limit": 3000,
            },
          ],
        ],
        "results": Array [
          Object {
            "isThrow": false,
            "value": Promise {},
          },
        ],
      }
    }
  >
    <Ripple
      borderColor="#ff5d00"
      borderStyle="solid"
      borderWidth="4px"
      height="64px"
      minHeight="80vh"
      width="64px"
    >
      <div
        className="lds-ripple-parent"
        style={
          Object {
            "minHeight": "80vh",
          }
        }
      >
        <div
          className="lds-ripple"
          style={
            Object {
              "height": "64px",
              "width": "64px",
            }
          }
        >
          <div
            style={
              Object {
                "borderColor": "#ff5d00",
                "borderStyle": "solid",
                "borderWidth": "4px",
              }
            }
          />
          <div
            style={
              Object {
                "borderColor": "#ff5d00",
                "borderStyle": "solid",
                "borderWidth": "4px",
              }
            }
          />
        </div>
      </div>
    </Ripple>
  </IrsPlan>
</Connect(IrsPlan)>
`;

exports[`containers/pages/IRS/plan renders correctly if prop planById is not null 1`] = `
<Connect(IrsPlan)
  history={
    Object {
      "action": "POP",
      "block": [Function],
      "createHref": [Function],
      "go": [Function],
      "goBack": [Function],
      "goForward": [Function],
      "length": 1,
      "listen": [Function],
      "location": Object {
        "hash": "",
        "pathname": "/",
        "search": "",
        "state": undefined,
      },
      "push": [Function],
      "replace": [Function],
    }
  }
  location={[MockFunction]}
  match={
    Object {
      "isExact": true,
      "params": Object {
        "id": "ed2b4b7c-3388-53d9-b9f6-6a19d1ffde1f",
      },
      "path": "/intervention/irs/plan/:id",
      "url": "/intervention/irs/plan/ed2b4b7c-3388-53d9-b9f6-6a19d1ffde1f",
    }
  }
  planById={
    Object {
      "id": "0230f9e8-1f30-5e91-8693-4c993661785e",
      "plan_date": "2019-08-09",
      "plan_effective_period_end": "2019-08-29",
      "plan_effective_period_start": "2019-08-09",
      "plan_fi_reason": "",
      "plan_fi_status": "",
      "plan_id": "0230f9e8-1f30-5e91-8693-4c993661785e",
      "plan_intervention_type": "IRS",
      "plan_jurisdictions_ids": Array [
        "3952",
      ],
      "plan_status": "draft",
      "plan_title": "IRS 2019-08-09",
      "plan_version": "1",
    }
  }
  supersetService={
    [MockFunction] {
      "calls": Array [
        Array [
          undefined,
          Object {
            "row_limit": 3000,
          },
        ],
      ],
      "results": Array [
        Object {
          "isThrow": false,
          "value": Promise {},
        },
      ],
    }
  }
>
  <IrsPlan
    allJurisdictionIds={Array []}
    assignmentsArray={Array []}
    fetchAllJurisdictionIdsActionCreator={[Function]}
    fetchAssignmentsActionCreator={[Function]}
    fetchJurisdictionsActionCreator={[Function]}
    fetchOrganizationsActionCreator={[Function]}
    fetchPlansActionCreator={[Function]}
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "createHref": [Function],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
    isDraftPlan={null}
    isFinalizedPlan={null}
    isNewPlan={false}
    jurisdictionsById={Object {}}
    loadedJurisdictionIds={Array []}
    location={[MockFunction]}
    match={
      Object {
        "isExact": true,
        "params": Object {
          "id": "ed2b4b7c-3388-53d9-b9f6-6a19d1ffde1f",
        },
        "path": "/intervention/irs/plan/:id",
        "url": "/intervention/irs/plan/ed2b4b7c-3388-53d9-b9f6-6a19d1ffde1f",
      }
    }
    organizationsById={Object {}}
    planById={
      Object {
        "id": "0230f9e8-1f30-5e91-8693-4c993661785e",
        "plan_date": "2019-08-09",
        "plan_effective_period_end": "2019-08-29",
        "plan_effective_period_start": "2019-08-09",
        "plan_fi_reason": "",
        "plan_fi_status": "",
        "plan_id": "0230f9e8-1f30-5e91-8693-4c993661785e",
        "plan_intervention_type": "IRS",
        "plan_jurisdictions_ids": Array [
          "3952",
        ],
        "plan_status": "draft",
        "plan_title": "IRS 2019-08-09",
        "plan_version": "1",
      }
    }
    planId="ed2b4b7c-3388-53d9-b9f6-6a19d1ffde1f"
    supersetService={
      [MockFunction] {
        "calls": Array [
          Array [
            undefined,
            Object {
              "row_limit": 3000,
            },
          ],
        ],
        "results": Array [
          Object {
            "isThrow": false,
            "value": Promise {},
          },
        ],
      }
    }
  >
    <Ripple
      borderColor="#ff5d00"
      borderStyle="solid"
      borderWidth="4px"
      height="64px"
      minHeight="80vh"
      width="64px"
    >
      <div
        className="lds-ripple-parent"
        style={
          Object {
            "minHeight": "80vh",
          }
        }
      >
        <div
          className="lds-ripple"
          style={
            Object {
              "height": "64px",
              "width": "64px",
            }
          }
        >
          <div
            style={
              Object {
                "borderColor": "#ff5d00",
                "borderStyle": "solid",
                "borderWidth": "4px",
              }
            }
          />
          <div
            style={
              Object {
                "borderColor": "#ff5d00",
                "borderStyle": "solid",
                "borderWidth": "4px",
              }
            }
          />
        </div>
      </div>
    </Ripple>
  </IrsPlan>
</Connect(IrsPlan)>
`;
